src/pages/index.js

```diff
@@ -72,41 +125,134 @@ function getCardElement(data) {
   cardImageEl.src = data.link;
   cardImageEl.alt = data.name;
 
-  cardLikeBtn.addEventListener("click", function () {
-    cardLikeBtn.classList.toggle("card__like-btn_liked");
-  });
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
 
-  cardDeleteBtn.addEventListener("click", function () {
-    cardElement.remove();
+  cardLikeBtn.addEventListener("click", () => {
+    if (cardLikeBtn.classList.contains("card__like-btn_liked")) {
+      api
+        .toggleLike(data._id, true)
+        .then(() => {
+          cardLikeBtn.classList.toggle("card__like-btn_liked");
+        })
+        .catch(console.error);
+    } else {
+      api
+        .toggleLike(data._id, false)
+        .then(() => {
+          cardLikeBtn.classList.toggle("card__like-btn_liked");
+        })
+        .catch(console.error);
+    }
   });
 
-  cardImageEl.addEventListener("click", function () {
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteBtn(cardElement, data._id)
+  );
+
+  cardImageEl.addEventListener("click", (evt) => {
+    evt.stopPropagation();
     openModal(previewModal);
+    previewModalImageEl.alt = data.name;
     previewModalImageEl.src = data.link;
-    previewModalImageEl.alt = data.alt;
     previewModalCaptionEl.textContent = data.name;
   });
 
   return cardElement;
 }
 
 function handleEditFormSubmit(evt) {
+  const editFormSubmitBtn = evt.submitter;
+  setButtonText(editFormSubmitBtn, true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions